test/http.js

```diff
@@ -175,7 +175,7 @@ describe('http', () => {
       // cross-fetch exposes FetchAPI methods onto global
       require('cross-fetch/polyfill')
       expect(global.Headers).toBeInstanceOf(Function)
-      const headers = new Headers() // eslint-disable-line no-undef
+      const headers = new Headers()
```

Comment:
2 things. First, I think we should move  `require('cross-fetch/polyfill')` to global top of test file. Second, we should not remove the `eslint-disable` because it will otherwise still be a linting problem, at least until we are able to import { Headers } directly.

Reply:
Regarding eslint-disable:
I've created the the global override for tests [here](https://github.com/swagger-api/swagger-js/pull/1523/files#diff-799dd7ade334108d9d2c335b978f7521R5). As we refer to `Header`, `Request` and `Response` global symbols routinely and I had to write `eslint-disable` more than 4 times it just made sense to make global override. But...I agree that change like this breaks current convention and should be systematic and reflected  everywhere instead having it locally in one file only. So I'll fall back to `eslint-disable` on a line basis and do global symbol allowance in separate PR and that

Reply:
Regarding `require('cross-fetch/polyfill')`

I do understand why applying the polyfill is made locally. The original author was trying to achieve complete isolation which I think makes total sense. But the complete isolation is not finalized because [afterEach hook](https://github.com/swagger-api/swagger-js/blob/b9189befb43294ebbdf3c384a8d94c8bfb41e080/test/http.js#L17) doesn't contain code to cleanup the polyfill symbols from global object (Request, Response, fetch, Headers...).

So we have following options:
**1.)** achieve complete isolation by applying polyfill in test bodies but finalize proper clearing of global object
**2.)** as you suggest move polyfilling to top of every file
**3.)** Import node-fetch symbols explicitly from cross-fetch

I'm more inclined for  **3.)** or **1.)** (in that order). Second options just doesn't feel right in context of unitests.

Reply:
I believe using `import fetch from cross-fetch` will work, but haven't had a chance to fully confirm or examine other possible artifacts. So let's try that for now.

Reply:
It will, but that option 3.) and as I said I don't want to make local changes to conventions but rather change the convention for entire codebase in one PR.